pkg/apiserver/user/sqlauth/sqlauth.go

```diff
@@ -50,6 +79,64 @@ func (a *Authenticator) Authenticate(f user.AuthenticateForm) (*utils.SessionUse
 	}
 	defer utils.CloseTiDBConnection(db) //nolint:errcheck
 
+	// Check privileges
+	// 1. Check whether TiDB SEM is enabled
+	resData, err := a.tidbClient.SendGetRequest("/config")
+	if err != nil {
+		return nil, user.ErrSignInOther.WrapWithNoMessage(err)
+	}
+	var config tidbSecurityConfig
+	err = json.Unmarshal(resData, &config)
+	if err != nil {
+		return nil, user.ErrSignInOther.WrapWithNoMessage(err)
+	}
+	// 2. Get grants
+	var grantRows []string
+	err = db.Raw("show grants for current_user()").Find(&grantRows).Error
+	if err != nil {
+		return nil, user.ErrSignInOther.WrapWithNoMessage(err)
+	}
+	grants := parseGrants(grantRows)
+	// 3. Check
+	// Following base privileges are required
+	// - ALL PRIVILEGES
+	// - or
+	// - PROCESS
+	// - SHOW DATABASES
+	// - SYSTEM_VARIABLES_ADMIN or SUPER (SUPER includes SYSTEM_VARIABLES_ADMIN)
+	// - CONFIG
+	// - DASHBOARD_CLIENT
```

Comment:
DASHBOARD_CLIENT is a sub privilege of SUPER.
I'm not sure whether CONFIG is a sub privilege of SUPER.

@crazycs520 Please provide necessary information.

Reply:
OK, @crazycs520 please help check, currently, the logic is following the design doc.

Reply:
`CONFIG` is not a sub privilege of `SUPER`.

Reply:
> `CONFIG` is not a sub privilege of `SUPER`.

What about the `DASHBOARD_CLIENT`?

Reply:
`DASHBOARD_CLIENT` also is not a sub privilege of `SUPER`.

Reply:
@crazycs520 DASHBOARD_CLIENT is a dynamic privilege, why is it not covered by SUPER?

Reply:
After confirm with @Morgan, `DASHBOARD_CLIENT` should be covered by SUPER